Python/_warnings.c

```diff
@@ -1146,50 +1151,36 @@ static PyMethodDef warnings_functions[] = {
 static PyObject *
 create_filter(PyObject *category, const char *action)
 {
-    static PyObject *ignore_str = NULL;
-    static PyObject *error_str = NULL;
-    static PyObject *default_str = NULL;
-    static PyObject *always_str = NULL;
-    PyObject *action_obj = NULL;
+    _Py_IDENTIFIER(ignore);
```

Comment:
This optimization doesn't make sense since `create_filter()` is used only few times at start. It would be better to pass `_Py_Identifier *` instead of `const char *`, this could save few lines of code.

Reply:
Sorry, I chose to put two changes into the same commit. This change was made to allow to free strings at Python finallization, it's not an optimization.